File: Exercise/Medium/06. Find the Missing Number in a Sequence/Code/Find the Missing Number in a Sequence.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class Main {
    public static int findMissingNumber(List<Integer> arr) {
        int expectedSum = (arr.get(0) + arr.get(arr.size() - 1)) * (arr.size() + 1) / 2;
        int actualSum = 0;
        for (int num : arr) {
            actualSum += num;
        }
        return expectedSum - actualSum;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        String line = scanner.nextLine();
        
        String[] tokens = line.split(",");
        List<Integer> arr = new ArrayList<>();
        for (String token : tokens) {
            arr.add(Integer.parseInt(token.trim()));
        }

        System.out.println(findMissingNumber(arr));
    }
}

File: Exercise/Medium/06. Find the Missing Number in a Sequence/Code/Find the Missing Number in a Sequence.py
import sys

def find_missing_number(arr):
    expected_sum = (arr[0] + arr[-1]) * (len(arr) + 1) / 2
    actual_sum = 0
    for num in arr:
        actual_sum += num
    return expected_sum - actual_sum

def main():
    line = input()

    arr = list(map(int, line.split(',')))

    print(find_missing_number(arr))

if __name__ == "__main__":
    main()